temporal-opentracing/src/main/java/io/temporal/opentracing/internal/OpenTracingContextPropagationUtils.java

```diff
@@ -0,0 +1,99 @@
+/*
+ *  Copyright (C) 2020 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ *  Copyright 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ *  use this file except in compliance with the License. A copy of the License is
+ *  located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package io.temporal.opentracing.internal;
+
+import com.google.gson.reflect.TypeToken;
+import io.opentracing.SpanContext;
+import io.opentracing.Tracer;
+import io.opentracing.propagation.Format;
+import io.opentracing.propagation.TextMap;
+import io.temporal.api.common.v1.Payload;
+import io.temporal.common.converter.DataConverter;
+import io.temporal.common.interceptors.Header;
+import java.lang.reflect.Type;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Optional;
+
+public class OpenTracingContextPropagationUtils {
+  private static final String tracerHeaderKey = "_tracer-data";
+
+  public static void writeSpanContextToHeader(
```

Comment:
Did you consider making these non-static and instantiating this accessor as part of init?

Reply:
I don't really see too much of a benefit here, this class doesn't really have anything it doesn't take as an input parameter. I can make it a helper if you really prefer, but the method signatures will be totally the same I think.

Reply:
I generally have a preference towards non-static helpers, they are less prone to misuse. But I'm okay leaving it as is.